effectful/handlers/pyro.py

```diff
@@ -330,29 +329,43 @@ def positional_distribution(
 ) -> tuple[TorchDistribution, Naming]:
     shape = d.shape()
     d = defterm(d)
-    dist_constr, args = d.args[0], d.args[1:]
 
-    if not (
-        d.op is call
-        and (
-            issubclass(dist_constr, TorchDistribution)
-            or issubclass(dist_constr, dist.torch_distribution.TorchDistributionMixin)
-        )
-    ):
+    if not isinstance(d, _DistributionTerm):
```

Comment:
A check of this form should not appear in a semantic function - behavior should not depend on specific `Term` subtypes that are internal implementation details of `defdata`/`defterm`.

Is there a syntactic test we can apply instead, since the previous version is no longer valid? Alternatively, can we make the previous version valid again?